src/tty.rs

```diff
@@ -147,6 +148,18 @@ fn set_controlling_terminal(fd: c_int) {
     }
 }
 
+/// Missing from FreeBSD libc crate?? Remove this case
+/// when TIOCSCTTY has been added to FreeBSD's libc crate.
+#[cfg(target_os = "freebsd")]
+fn set_controlling_terminal(fd: c_int) {
+    const TIOCSCTTY: libc::c_ulong = 536900705;
```

Comment:
Instead of duplicating this entire function, could we conditionally import or define TIOCSCTTY at the top of this file?